src/applications/common/schemaform/fields/AutosuggestField.jsx

```diff
@@ -85,11 +101,21 @@ export default class AutosuggestField extends React.Component {
       return suggestion.id;
     }
 
+    // When freeInput is true, we'll return the label to the api instead of the id
+    if (this.props.uiSchema['ui:options'].freeInput) {
+      return suggestion.label;
+    }
+
     return _.set('widget', 'autosuggest', suggestion);
   }
 
   handleInputValueChange = (inputValue) => {
     if (inputValue !== this.state.input) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      if (uiOptions.queryForResults) {
```

Comment:
It feels like in this case, if we're calling a server for results, all of the matching logic would be offloaded there, and I'm not sure how much of the rest of the method makes sense in that case.

Reply:
This wasn't a deliberate decision when I wrote it, but I think the rest of the method is still useful for while we're waiting for `getOptions` to fire (because debounce). While we're still typing, it'll match (or not) using our current list. When we stop typing, it'll fetch the new list based on that input.